nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/StandardNiFiServiceFacade.java

```diff
@@ -5357,6 +5355,8 @@ public RegisteredFlow registerVersionedFlow(final String registryId, final Regis
         }
 
         try {
+            final String generatedId = registry.generateFlowId(flow.getName());
```

Comment:
This might be something I would prefer not to expose outside of the node. This should be an internal implementation detail for the registry node (or implementation). Is there any particular reason this leaves the boundaries of a registry?

Reply:
Previously the framework generated a UUID in service facade before calling the registerFlow method in registry client, but for GitHub we don't want a UUID, so this just allows the framework to ask the client to generate the id, and the default behavior can return a UUID like before, and the GitHub impl can do something different.